packages/network-controller/src/NetworkController.ts

```diff
@@ -708,118 +772,125 @@ export class NetworkController extends BaseControllerV2<
+   * @param options - Additional configuration options.
+   * @param options.referrer - Used to create a metrics event; the site from which the call originated, or 'metamask' for internal calls.
+   * @param options.source - Used to create a metrics event; where the event originated (i.e. from a dapp or from the network form).
+   * @param options.setActive - If true, switches to the network upon adding or updating it (default: false).
+   * @returns The ID for the added or updated network configuration.
    */
   async upsertNetworkConfiguration(
-    { rpcUrl, chainId, ticker, nickname, rpcPrefs }: NetworkConfiguration,
+    networkConfiguration: NetworkConfiguration,
     {
-      setActive = false,
       referrer,
       source,
-    }: { setActive?: boolean; referrer: string; source: string },
+      setActive = false,
+    }: {
+      referrer: string;
+      source: string;
+      setActive?: boolean;
+    },
   ): Promise<string> {
-    assertIsStrictHexString(chainId);
+    const sanitizedNetworkConfiguration = (
+      ['rpcUrl', 'chainId', 'ticker', 'nickname', 'rpcPrefs'] as const
+    ).reduce((obj, key) => {
+      if (key in networkConfiguration) {
+        return { ...obj, [key]: networkConfiguration[key] };
+      }
+      return obj;
+    }, {} as NetworkConfiguration);
+    const { rpcUrl, chainId, ticker } = sanitizedNetworkConfiguration;
 
+    assertIsStrictHexString(chainId);
     if (!isSafeChainId(chainId)) {
       throw new Error(
         `Invalid chain ID "${chainId}": numerical value greater than max safe value.`,
       );
     }
-
     if (!rpcUrl) {
       throw new Error(
         'An rpcUrl is required to add or update network configuration',
       );
     }
-
     if (!referrer || !source) {
       throw new Error(
         'referrer and source are required arguments for adding or updating a network configuration',
       );
     }
-
     try {
       new URL(rpcUrl);
     } catch (e: any) {
       if (e.message.includes('Invalid URL')) {
         throw new Error('rpcUrl must be a valid URL');
       }
     }
-
     if (!ticker) {
       throw new Error(
         'A ticker is required to add or update networkConfiguration',
       );
     }
 
-    const newNetworkConfiguration = {
-      rpcUrl,
-      chainId,
-      ticker,
-      nickname,
-      rpcPrefs,
-    };
-
-    const oldNetworkConfigurations = this.state.networkConfigurations;
+    const autoManagedNetworkClientRegistry =
+      this.#ensureAutoManagedNetworkClientRegistryPopulated();
 
-    const oldNetworkConfigurationId = Object.values(
-      oldNetworkConfigurations,
-    ).find(
-      (networkConfiguration) =>
-        networkConfiguration.rpcUrl?.toLowerCase() === rpcUrl?.toLowerCase(),
-    )?.id;
+    const existingNetworkConfiguration = Object.values(
+      this.state.networkConfigurations,
+    ).find((nc) => nc.rpcUrl.toLowerCase() === rpcUrl.toLowerCase());
```

Comment:
Nit: Maybe we could use the parameter name `networkConfig` instead? Generally we avoid 1-2 letter variable names (this is one of the lint rules in the updated MetaMask ESLint base config)

Reply:
Sure, updated here: [85c7e9c](https://github.com/MetaMask/core/pull/1439/commits/85c7e9c221012c8822c1cddebf9467736dbe60b7)